### PR TITLE
Issue 6 question

### PR DESCRIPTION
**Pull request by barzerman**

### CODE REVIEW

File: lib/barzer/barzer_objects.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    @classmethod
+    def make_bead_from_dict(cls, data, default_type=None):
+        the_type = data.get('type', default_type)
+        if the_type == 'entlist':
+            return [Entity(x) for x in data['data']]
+
+        bead_type = cls.NAME_TYPE.get(the_type)
+        if bead_type:
+            try:
+                return bead_type(data) if bead_type else None
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py]: I suppose just return bead_type(data) here as we already checked that bead_type is not None

File: lib/calc_graph.py [hunk 2]
```diff
@@ -148,33 +173,34 @@ def find_next_unset_child(self):
...
     def to_dict(self):
-        data = {'op': self.op.op_name, 'value': self.value.to_dict()}
+        data = {'op': self.op.op_name if self.op else CGOperator.op_name, 'value': self.value.to_dict()}
         if self.children:
             data['children'] = [c.to_dict() for c in self.children]
         return data
 
+    def op_calc(self, children, input_val):
+        return (self.op.calc(children, input_val)
+                if self.op else CGOperator.static_calc(
```
Comment (0-1-0) [inline lib/calc_graph.py]: imho, default op=CGOperator will be more clear than considering CGNodes with possible op=None

File: lib/barzer/barzer_objects.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+    def value_str(self):
+        return str(self.value)
+
+class Token(ValueBead):
+    """ regular token """
+    type_name = 'token'
+
+    def value_str(self):
+        return "'"+str(self.value)+"'"
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py]: add spaces around arithmetic operator

File: lib/barzer/barzer_objects.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            self.id == ent.id) # pylint: disable=no-member
+
+class Entity(EntityBase):
+    def __init__(self, data):
+        """
+        Arguments:
+            data (dict) - dictionary returned from barzer
+        """
+        super(Entity, self).__init__(data)
+        self.eclass = data['class']
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py:63]: why not .get(..) here?

File: lib/barzer/barzer_objects.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+            lo = the_type(self.lo)
+        except Exception:
+            lo = None
+
+        try:
+            hi = the_type(self.lo)
+        except Exception:
+            hi = None
+
+        if lo is not None and hi is not None:
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py:106]: if lo and hi ?

File: lib/barzer/barzer_objects.py [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        return '({}, {})'.format(str(self.lo), str(self.hi))
+
+    def get_as_type_pair(self, the_type=float):
+        try:
+            lo = the_type(self.lo)
+        except Exception:
+            lo = None
+
+        try:
+            hi = the_type(self.lo)
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py:102]: hi = the_type(self.hi)

File: lib/barzer/barzer_objects.py [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    type_name = "time"
+    """ time of day """
+
+
+class Timestamp(Bead):
+    type_name = "timestamp"
+    """ """
+    def __init__(self, data):
+        super(Timestamp, self).__init__(data)
+        self.date, self.time = data['date'], data['time']
```
Comment (0-1-0) [inline lib/barzer/barzer_objects.py:177]: .get('date'), .get('time) ?

Comment (0-1-0) [inline lib/barzer/barzer_objects.py:177]: why not to convert to python Date?

File: lib/barzer/barzer_objects.py (hunk 4 above)
Comment (barzerman) [inline lib/barzer/barzer_objects.py:63]: because it's supposed to be a valid entity

File: lib/barzer/barzer_objects.py (hunk 5 above)
Comment (barzerman) [inline lib/barzer/barzer_objects.py:106]: no. 0 != None

File: lib/barzer/barzer_objects.py (hunk 7 above)
Comment (barzerman) [inline lib/barzer/barzer_objects.py:177]: valid Timestamp must have both `date` and `time` so no, not `get`
as to converting to `datetime` - that's a separate option but I want to preserve the fact that it's the bead of type `Date`, `Time` or `DateTime`

File: lib/calc_graph.py (hunk 2 above)
Comment (barzerman) [inline lib/calc_graph.py]: not sure why - but doing that fucks up the test - too lazy now. will figure out later .  In general though it should be fine to have `None` op

**Pull request merged**